browser/ui/views/profiles/brave_avatar_toolbar_button.cc

```diff
@@ -38,26 +39,53 @@ namespace {
 constexpr int kHighlightRadius = 36;
 constexpr int kBraveAvatarButtonHorizontalSpacing = 10;
 
-class BraveToolbarButtonHighlightPathGenerator
+class BraveAvatarButtonHighlightPathGenerator
     : public views::HighlightPathGenerator {
  public:
-  // HighlightPathGenerator:
-  SkPath GetHighlightPath(const views::View* view) override {
-    gfx::Rect rect(view->size());
-    rect.Inset(GetToolbarInkDropInsets(view));
-    SkPath path;
-    path.addRoundRect(gfx::RectToSkRect(rect), kHighlightRadius,
-                      kHighlightRadius);
-    return path;
+  explicit BraveAvatarButtonHighlightPathGenerator(
+      const base::WeakPtr<BraveAvatarToolbarButton>& avatar_button)
+      : avatar_button_(avatar_button) {}
+  BraveAvatarButtonHighlightPathGenerator(
+      const BraveAvatarButtonHighlightPathGenerator&) = delete;
+  BraveAvatarButtonHighlightPathGenerator& operator=(
+      const BraveAvatarButtonHighlightPathGenerator&) = delete;
+  ~BraveAvatarButtonHighlightPathGenerator() override = default;
+
+  absl::optional<gfx::RRectF> GetRoundRect(const gfx::RectF& bounds) override {
+    gfx::Rect rect(avatar_button_->size());
+    rect.Inset(GetToolbarInkDropInsets(avatar_button_.get()));
+    DCHECK(avatar_button_);
+    if (avatar_button_->GetAvatarButtonState() ==
+        AvatarToolbarButton::State::kAnimatedUserIdentity) {
+      // When it's animating we shouldn't draw rounded highlight as the button
+      // has different radius. We draw rect here then the button will clip the
+      // highlight properly.
+      return gfx::RRectF(gfx::RectF(rect), 0);
```

Comment:
It seems clipping doesn't work as expected like below.
<img width="152" alt="Screenshot 2023-01-15 at 11 32 33 AM" src="https://user-images.githubusercontent.com/6786187/212520203-d34e24a8-8a5f-4118-88b4-202034670195.png">
I think setting same radius like below would make more better?
```
      return gfx::RRectF(gfx::RectF(rect),
                         ChromeLayoutProvider::Get()->GetCornerRadiusMetric(
                             views::Emphasis::kMaximum, {}));
```

Reply:
Yeah, sounds good! will try.

Reply:
Looks working great! Thanks.